src/mixins/prevent-unsaved-mixin.ts

```diff
@@ -6,38 +6,86 @@ export const PreventUnsavedMixin = <T extends Constructor<LitElement>>(
   superClass: T
 ) =>
   class extends superClass {
+    private _historyState: string | null = null;
+
+    private _listenersAttached = false;
+
+    private _isNavigating = false;
+
     private _handleClick = async (e: MouseEvent) => {
-      // get the right target, otherwise the composedPath would return <home-assistant> in the new event
-      const target = e.composedPath()[0];
-      if (!isNavigationClick(e)) {
+      const href = isNavigationClick(e);
+      if (!href) {
         return;
       }
 
+      const target = e.composedPath()[0];
```

Comment:
This comment is still valid

```suggestion
      if (!isNavigationClick(e)) {
        return;
      }
      // get the right target, otherwise the composedPath would return <home-assistant> in the new event
      const target = e.composedPath()[0];
```